packages/dds/matrix/src/matrix.ts

```diff
@@ -521,20 +524,31 @@ export class SharedMatrix<T = any>
 			SnapshotPath.cols,
 			this.cols.summarize(this.runtime, this.handle, serializer),
 		);
+		const cellsSnapshot = this.cells.snapshot();
+		const props: Record<string, number> = {
+			csSize: cellsSnapshot.length, // cellsSnapshotSize
+			rc: this.rowCount, // rowCount
+			cc: this.colCount, // colCount
+		};
 		const artifactsToSummarize = [
-			this.cells.snapshot(),
+			cellsSnapshot,
 			this.pending.snapshot(),
 			this.setCellLwwToFwwPolicySwitchOpSeqNumber,
 		];
 
 		// Only need to store it in the snapshot if we have switched the policy already.
 		if (this.setCellLwwToFwwPolicySwitchOpSeqNumber > -1) {
-			artifactsToSummarize.push(this.cellLastWriteTracker.snapshot());
+			const trackerMatrixSnapshot = this.cellLastWriteTracker.snapshot();
+			artifactsToSummarize.push(trackerMatrixSnapshot);
+			props.tmsSize = trackerMatrixSnapshot.length; // trackerMatrixSnapshotSize
 		}
 		builder.addBlob(
 			SnapshotPath.cells,
 			serializer.stringify(artifactsToSummarize, this.handle),
 		);
+		if (telemetryContext?.push !== undefined) {
+			telemetryContext.push("fluid:SharedMatrix", "details", [JSON.stringify(props)]);
```

Comment:
Curious, why it has to be stringified? Can it be same object? I.e. stringification to be done once, by context, on top of array of objects?

Reply:
Down below, type is value: TelemetryEventPropertyType[]):, so that should be natural step, no?

Reply:
We took a step of supporting single-level arrays and objects. Object nested in array isn't supported, but could be if needed.

Like here it looks like you're doing just that - logging an array of records.  How are you expecting these to be consumed?  e.g. for our 1P partners who share their telemetry with us, in Azure DataExplorer - are you going to navigate through the dynamic object and do some aggregation across a whole matrix per summary?  Or what?